slice-core/src/main/java/com/cognifide/slice/core/internal/monitoring/SliceStatistics.java

```diff
@@ -0,0 +1,87 @@
+/*-
+ * #%L
+ * Slice - Core
+ * %%
+ * Copyright (C) 2012 Cognifide Limited
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.cognifide.slice.core.internal.monitoring;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+import java.util.Map;
+import java.util.TreeMap;
+
+import org.apache.commons.lang.StringUtils;
+
+import com.cognifide.slice.api.injector.InjectorConfig;
+import com.cognifide.slice.core.internal.injector.InjectorHierarchy;
+import com.google.inject.Injector;
+
+public class SliceStatistics {

```

Comment:
I believe we could make this class a OSGi service.
It is used by the console service only so it would work.
It would also simplify method signatures as we would inject InjectorHierarchy instead of passing it through parameters.